src/runtimes/node/bundlers/esbuild/plugin_native_modules.ts

```diff
@@ -39,6 +39,10 @@ const getNativeModulesPlugin = (externalizedModules: NativeNodeModules): Plugin
 
       if (!pkg) return
 
+      if (pkg[1].startsWith('node:')) {
```

Comment:
Note that "native modules" in this context means Node modules with native binaries, whereas the `node:` prefix denotes built-in Node modules.

By having these together, I worry that making changes to the native module detection might break the built-in module handling in the future.

Have you considered creating a separate plugin for this? Something like:

```js
build.onResolve({ filter: /^node: }, async (args) => {
  return { external: true }
})
```